general/1_terminal/Leonid-Kharin/task4.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash jq
+
+file=$(cat result.json)
+
+echo "please, wait 15 seconds"
+
+# iterate over all keys with value text
+
+for ((i=0; i >(-1); i++)) do 
+
+   text=$(echo $file | jq '.messages'[$i] | jq '.text' ) 
+
+   if [[ $text == "null" ]]; then #  when the list ends the value becomes null
+      break
+   fi
+   
+   if [[ $text == *"https"* ]]; then
+      temp=10 
+   else
+       echo $text >> r.json # action that don't matter
+   fi
+
+done
```

Comment:
I like what you did here: this is pretty much how you would solve this in Python. 
Let's try to rewrite it a bit.

Reply:
What I mean is to try to rewrite it without using a loop, or at least using a `while` loop.